src/main/java/seedu/gamebook/model/util/GameEntriesDateComparator.java

```diff
@@ -10,6 +10,7 @@
 public class GameEntriesDateComparator implements Comparator<GameEntry> {
     @Override
     public int compare(GameEntry gameOne, GameEntry gameTwo) {
+        assert gameOne != null && gameTwo != null;
```

Comment:
```suggestion
        requireAllNonNull(gameOne, gameTwo);
```

`seedu.gamebook.commons.util.CollectionUtil.requireAllNonNull` offers this functionality.

Reply:
requireNonNull() or requireAllNonNull() have already been called in various places in GameEntryList, such as in GameEntryList#add(), GameEntryList#setGameEntry(), GameEntryList#setGameEntries(). Therefore, the assertions are simply to verify these preconditions.